### PR TITLE
task (deprecation): remove deprecated methods for 3.0

### PR DESCRIPTION
**Pull request by george-bafaloukas-forgerock**
- Removed getAuthorizeUrl method
- Removed createVerifier parameter
- Removed nonce utility function
- Changed the default behaviour in case of unidentified storage, to be  the localStorage option

### CODE REVIEW

File: src/token-manager/index.ts [hunk 1]
```diff
@@ -124,23 +124,17 @@ abstract class TokenManager {
     try {
       let parsedUrl;
 
-      // Check expected browser support
-      if (support === 'legacy' || support === undefined) {
-        // To support legacy browsers, iframe works best with short timeout
-        parsedUrl = new URL(await OAuth2Client.getAuthorizeUrl(authorizeUrlOptions));
-      } else {
```
Comment (cerebrl) [inline src/token-manager/index.ts:131]: It's important that this is kept. We still need to be able to support the legacy `iframe` method for calling the `authorize` endpoint. So, that means the functionality of `getAuthorizeUrl` needs to be preserved, but the name is what's replaced with a more semantic name.

File: src/oauth2-client/index.ts [hunk 2]
```diff
@@ -72,59 +72,6 @@ abstract class OAuth2Client {
     return url.toString();
   }
 
-  /**
-   * DEPRECATED
-   * Calls the authorize URL with an iframe. If successful,
-   * it returns the callback URL with authentication code,
-   * optionally using PKCE.
-   */
-  public static async getAuthorizeUrl(options: GetAuthorizationUrlOptions): Promise<string> {
```
Comment (cerebrl) [inline src/oauth2-client/index.ts:81]: This function needs to be preserved, but we need to rename it. The problem with this name was that it was misleading. The calling of this method does not get the authorize URL itself and return it. The calling of this method returns the result of the request to `/authorize`. So, we could name it something like `authorizeByIframe`, and I think that would clear this confusion up. Thoughts?

Comment (george-bafaloukas-forgerock) [reply src/oauth2-client/index.ts:81]: OK Now it makes more sense. I was going with the doc here:
https://docs.google.com/spreadsheets/d/1_TW6F55PAp52eOqUoZIWYyR7MSDDyEg3fs-v8oWuMEc/edit#gid=0
That said remove, so went and nuked it from orbit.

"authorizeByIframe" or "authorizeWithIframe", "authorizeForLegacy"

All sound good to me. Will revisit and reinstate that. Please let me know what is your preference for name

Comment (cerebrl) [reply src/oauth2-client/index.ts:81]: Sorry, that doc should have said "rename". That's my bad.

As for names, either of the first two you mentioned works for me. There's also `getAuthCodeByIframe` or `getAuthCodeLegacy` that could work. I do like how `getAuthCode` is much more specific in its purpose. I'll leave it to you to name it :)

**Comment by spetrov** [comment]
Run tests

File: src/index.ts [hunk 3]
```diff
@@ -68,13 +68,11 @@ import TokenManager, { GetTokensOptions } from './token-manager';
 import TokenStorage from './token-storage';
 import UserManager from './user-manager';
 import Deferred from './util/deferred';
-import nonce from './util/nonce';
 import PKCE from './util/pkce';
 import LocalStorage from './util/storage';
 
 export {
   defaultMessageCreator,
-  nonce,
```
Comment (rodrigoareis) [inline src/index.ts:77]: Missing Copyright update

File: src/oauth2-client/index.ts [hunk 4]
```diff
@@ -73,14 +73,15 @@ abstract class OAuth2Client {
   }
 
   /**
-   * DEPRECATED
    * Calls the authorize URL with an iframe. If successful,
    * it returns the callback URL with authentication code,
    * optionally using PKCE.
+   * Method renamed in v3.
```
Comment (rodrigoareis) [inline src/oauth2-client/index.ts:79]: Missing Copyright update

File: src/token-manager/index.ts [hunk 5]
```diff
@@ -127,7 +127,7 @@ abstract class TokenManager {
       // Check expected browser support
       if (support === 'legacy' || support === undefined) {
         // To support legacy browsers, iframe works best with short timeout
-        parsedUrl = new URL(await OAuth2Client.getAuthorizeUrl(authorizeUrlOptions));
+        parsedUrl = new URL(await OAuth2Client.getAuthCodeByIframe(authorizeUrlOptions));
```
Comment (rodrigoareis) [inline src/token-manager/index.ts:130]: Missing Copyright update

File: src/token-storage/index.ts [hunk 6]
```diff
@@ -37,8 +37,8 @@ abstract class TokenStorage {
       // User supplied token store
       return await tokenStore.get(clientId);
     }
-    // if tokenStore is undefined, default to previous behavior
-    return await IndexedDBWrapper.get(clientId);
+    // if tokenStore is undefined, default to localStorage
```
Comment (rodrigoareis) [inline src/token-storage/index.ts:40]: Missing Copyright update

File: src/token-storage/token-storage.test.ts [hunk 7]
```diff
@@ -20,6 +20,7 @@ import { DB_NAME, TOKEN_KEY } from './constants';
 
 const config: ConfigOptions = {
   clientId: 'mockClientId',
+  tokenStore: 'indexedDB'
```
Comment (rodrigoareis) [inline src/token-storage/token-storage.test.ts:23]: Missing Copyright update

File: src/util/pkce.ts [hunk 8]
```diff
@@ -23,11 +23,8 @@ abstract class PKCE {
   /**
    * Creates a random verifier.
    */
-  public static createVerifier(num?: number): string {
-    if (num) {
-      console.warn('Deprecation warning: the parameter for `createVerifier` will be removed in v3');
-    }
-    return this.createRandomString(num || 32);
+  public static createVerifier(): string {
```
Comment (rodrigoareis) [inline src/util/pkce.ts:26]: Missing Copyright update

**Review by rodrigoareis — CHANGES_REQUESTED** [review]
All changes looks good to me. Just need to update the copyright for changed files

**Review by rodrigoareis — DISMISSED** [review]
Changes looks good to me

**Review by cerebrl — APPROVED** [review]
Looks good. Commits just need to be signed.

**Comment by cerebrl** [comment]
Also, now that this is close to merging, can you squash all the commits to one? That helps keep the Git log clean.

**Comment by george-bafaloukas-forgerock** [comment]
Yeah will try to squash and sign. Signing was supposed to be enabled on those...no idea what happened there

File: CHANGELOG.md [hunk 9]
```diff
@@ -14,13 +14,17 @@
 
 - Arbitrary query parameters now passed along to authorize URL creation
 - Fixed build issue when using Windows PowerShell
-- WebAuthn error handling is now standardized according to the WebAuthn spec
-- When WebAuthn encounters an error, it formats it appropriately and injects it into the hiddenCallback, allowing the developer to just send to AM or handle it specially (error is still thrown)
+- WebAuthn error handling is not standardized according to the WebAuthn spec
+- When WebAuthn encounters an error, it formats is appropriately and injects it into the hiddenCallback, allowing the developer to just send to AM or handle it specially.
```
Comment (cerebrl) [inline CHANGELOG.md:18]: It looks like `(error is still thrown)` got removed from this item.

File: src/oauth2-client/index.ts.rej [hunk 10]
```diff
@@ -0,0 +1,10 @@
+diff a/src/oauth2-client/index.ts b/src/oauth2-client/index.ts	(rejected hunks)
```
Comment (cerebrl) [inline src/oauth2-client/index.ts.rej:1]: Another rogue `.rej` file :)

Comment (george-bafaloukas-forgerock) [reply src/oauth2-client/index.ts.rej]: Should be fixed now

File: CHANGELOG.md (hunk 9 above)
Comment (george-bafaloukas-forgerock) [reply CHANGELOG.md]: fixed

**Review by cerebrl — APPROVED** [review]
Looks good. Ready for merge.

**Pull request merged**